src/support.h

```diff
@@ -61,8 +61,12 @@ inline static v8::MaybeLocal<T> ptr_to_maybe_local(T* ptr) {
 }
 
 template <class T>
-inline static T maybe_to_value(v8::Maybe<T> maybe, T default_value) {
-  return maybe.FromMaybe(default_value);
+inline static T* maybe_to_ptr(v8::Maybe<T> maybe) {
+  T* ptr = new T();
```

Comment:
This looks like a memory leak. When is this thing deleted?

Reply:
Probably never... I added delete before assigning, is that correct? I need to sharpen my cpp 😅

Reply:
Okay after some pondering I think `Maybe<T>` should be rewritten. 

In V8 APIs `Maybe<T>` is coupled with `V8_WARN_UNUSED_RESULT`. That suggests that `Maybe<T>` must always be consumed (or have `#[must_use]` attribute in Rust). 

So I guess I should return pointer to `Maybe<T>` to Rust and then consume its methods in Rust idiomatic way.

Reply:
This is not going in the right direction. Let me take a stab at it.